base/applications/rapps/include/rosui.h

```diff
@@ -302,54 +300,78 @@ class CUiBox
         *newRect = currentRect;
     }
 
-
 public:
+    //virtual ~CUiBox() {} // Needed for current CUiCollection implementation.
+
+    // Override in subclass
     virtual VOID ComputeMinimalSize(SIZE* size)
     {
-        // Override in subclass
         size->cx = max(size->cx, 0);
         size->cy = min(size->cy, 0);
-    };
+    }
 
+    // Override in subclass
     virtual VOID ComputeContentBounds(RECT* rect)
     {
-        // Override in subclass
-    };
+    }
 
-    virtual DWORD_PTR CountSizableChildren()
+    // Override in subclass
+    virtual INT CountSizableChildren()
     {
-        // Override in subclass
         return 0;
-    };
+    }
 
+    // Override in subclass
     virtual HDWP OnParentSize(RECT parentRect, HDWP hDwp)
     {
-        // Override in subclass
-        return NULL;
-    };
+        /* Pass hDwp down the call chain */
+        return hDwp;
+    }
 };
 
-class CUiPrimitive
+
+
+#if 1
+//
+// This is actually an interface of sorts...
+// Actually it could be helpful if it's a reference count
+// so that the UiCollection could implement the ref uncount and delete if needed.
+//
+class CUiPrimitive : public CUiBox
```

Comment:
This is related also to what the `CUiCollection` / `CUiContainer` classes can contain: What is, in this context, a "UI primitive", and why would instances of it be present in these collections/containers that are used by e.g. `CUiPanel` , if some of these are not measurable ?
Does making these inherit from `CUiBox` makes sense, or does there exist some situations where one would not want that ?
Cc @gigaherz for some clues about this.